build.gradle.kts

```diff
@@ -0,0 +1,201 @@
+}
+
+repositories {
+    mavenCentral()
+    jcenter()
+}
+
+tasks.withType<Wrapper>().configureEach {
+    gradleVersion = "5.0"
+    distributionType = Wrapper.DistributionType.ALL
+}
+
+fun javaSubprojects(action: Project.() -> Unit) {
+    subprojects.minus(project(":ui:linuxLauncher")).forEach { project ->
+        project.action()
+    }
+}
+
+var grgit: Grgit? = null
+if (rootProject.file(".git").exists()) {
+    project.apply {
+        plugin("org.ajoberstar.grgit")
+    }
+    grgit = Grgit.open()
+}
+
+javaSubprojects {
+    apply {
+        plugin("java")
+        plugin("org.gradle.jacoco")
+        plugin("org.gradle.pmd")
+        plugin("org.gradle.checkstyle")
+        plugin("net.ltgt.errorprone")
+        plugin("com.github.spotbugs")
+    }
+    repositories {
+        mavenCentral()
+        jcenter()
+        maven {
+            name = "WPILib Maven Release"
+            setUrl("http://first.wpi.edu/FRC/roborio/maven/release")
+        }
+        maven {
+            name = "rosjava Maven"
+            setUrl("https://github.com/rosjava/rosjava_mvn_repo/raw/master")
+        }
+        maven {
+            name = "GRIP ROS Maven"
+            setUrl("https://github.com/WPIRoboticsProjects/rosjava_mvn_repo/raw/master")
+        }
+    }
+
+    version = getVersionName()
+
+    dependencies {
+        "compile"(group = "com.google.code.findbugs", name = "annotations", version = "3.0.1")
+        "testCompile"(group = "net.jodah", name = "concurrentunit", version = "0.4.2")
+        "testCompile"(group = "org.hamcrest", name = "hamcrest-all", version = "1.3")
+        "testCompile"(group = "junit", name = "junit", version = "4.12")
+        "testCompile"(group = "com.google.truth", name = "truth", version = "0.34")
+        "testCompile"(group = "com.google.guava", name = "guava-testlib", version = "22.0")
+    }
+
+    checkstyle {
+        configFile = rootDir.resolve("checkstyle.xml")
+        toolVersion = "6.19"
+        if (project.hasProperty("ignoreCheckstyle")) {
+            isIgnoreFailures = true
+        }
+    }
+
+    pmd {
+        toolVersion = "5.6.0"
+        isConsoleOutput = true
+        val projectSourcesSets = this@javaSubprojects.sourceSets
+        sourceSets = listOf(projectSourcesSets["main"], projectSourcesSets["test"])
+        reportsDir = buildDir.resolve("reports/pmd")
+        ruleSetFiles = files(rootDir.resolve("pmd-ruleset.xml"))
+    }
+
+    configurations["errorprone"].apply {
+        resolutionStrategy.force("com.google.errorprone:error_prone_core:2.3.2")
+    }
+
+    spotbugs {
+        toolVersion = "3.1.7"
+        val javaSources = this@javaSubprojects.sourceSets
+        sourceSets = setOf(javaSources["main"], javaSources["test"])
+        excludeFilter = file("$rootDir/findBugsSuppressions.xml")
+        effort = "max"
+    }
+
+    tasks.withType<SpotBugsTask> {
+        reports {
+            xml.isEnabled = false
+            emacs.isEnabled = true
+        }
```

Comment:
You can get spotbugs to produce an `html` report if we don't want to deal with the emacs one anymore.